src/coreclr/vm/threadstatics.cpp

```diff
@@ -799,6 +803,34 @@ bool CanJITOptimizeTLSAccess()
     // Optimization is disabled for FreeBSD/arm64
 #elif defined(FEATURE_INTERPRETER)
     // Optimization is disabled when interpreter may be used
+#elif !defined(TARGET_OSX) && defined(TARGET_UNIX) && defined(TARGET_ARM64)
+    // Optimization is enabled for linux/arm64 only for static resolver.
+    // For static resolver, the TP offset is same for all threads.
+    // For dynamic resolver, TP offset returned is for the current thread and
+    // will be different for the other threads.
+    uint32_t* resolverAddress = reinterpret_cast<uint32_t*>(GetTLSResolverAddress());
+    int ip = 0;
+    if ((resolverAddress[ip] == 0xd503201f) || (resolverAddress[ip] == 0xd503241f))
+    {
+        // nop might not be present in older resolver, so skip it.
+
+        // nop or hint 32
```

Comment:
```suggestion
        // nop or bti
```

Reply:
Since "hint 32" is an instruction and I preferred having instruction in the comment, should I keep it as-is?

Reply:
[`bti`](https://developer.arm.com/documentation/ddi0596/2021-09/Base-Instructions/BTI--Branch-Target-Identification-#:~:text=A%20BTI%20instruction%20is%20used%20to%20guard%20against,region%2C%20a%20BTI%20instruction%20executes%20as%20a%20NOP.) is a more self-description name for the same instruction.

`hint` is reserved instruction encoding space. They are assigning concrete meaning and names to the instructions in this space as the architecture evolves. BTW: `nop` is same as `hint 0`.